plugins/module_utils/redfish_utils.py

```diff
@@ -3394,3 +3394,77 @@ def get_hpe_fan_percent_min(self):
+        if storage_uri is None:
+            return {'ret': False, 'msg': 'Storage resource not found'}
+
+        # Get Storage Collection
+        response = self.get_request(self.root_uri + storage_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        # Collect Storage Controllers
+        self.storage_controlllers_uris = [i['@odata.id'] for i in response['data'].get('Members', [])]
+        if not self.storage_controlllers_uris:
+            return {
+                'ret': False,
+                'msg': "StorageCollection's Members array is either empty or missing"}
+
+        # Check if storage controller supports volume creation
+        response = self.get_request(self.root_uri + self.storage_controlllers_uris[0])
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        if data.get('Volumes'):
+            response = self.get_request(self.root_uri + data['Volumes']['@odata.id'])
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            if data.get('@Redfish.CollectionCapabilities') and data['@Redfish.CollectionCapabilities'].get('Capabilities') \
+                and data['@Redfish.CollectionCapabilities']['Capabilities'][0].get('UseCase') \
+                    and data['@Redfish.CollectionCapabilities']['Capabilities'][0]['UseCase'] == "VolumeCreation":
+                self.storage_controller = self.storage_controlllers_uris[0]
+            else:
+                return {
+                    'ret': False,
+                    'msg': "Storage Controller does not support Volume creation"}
+
+        # Validate input parameters
+        required_parameters = ['RAIDType', 'Links']
+        allowed_parameters = ['CapacityBytes', 'DisplayName', 'InitializeMethod', 'MediaSpanCount',
+                              'Name', 'ReadCachePolicy', 'StripSizeBytes', 'VolumeUsage', 'WriteCachePolicy']
+
+        for parameter in required_parameters:
+            if not volume_details.get(parameter):
+                return {
+                    'ret': False,
+                    'msg': "%s are reuired parameter to create a volume" % str(required_parameters)}
+            elif parameter == "Links":
+                if not volume_details[parameter].get('Drives'):
+                    return {
+                        'ret': False,
+                        'msg': "'Drives' are reuired key under 'Links' to create a volume"}
+
+        # Navigate to the volume uri of the correct controller
+        response = self.get_request(self.root_uri + self.storage_controller)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        # Construct payload and issue POST command to create volume
+        payload = volume_details
+        response = self.post_request(self.root_uri + data['Volumes']['@odata.id'], payload)
```

Comment:
There will be cases where a Task is returned since it may not be possible to create the volume immediately; a system restart may also be required for the Task to run. We may need to add something here to handle the Task cases...

Reply:
Looks like no reboot is needed for Gen11 servers to reflect volume creation. Attaching the POST response for your reference. The response shows "Success" and we don't see any message for "Reboot Required", which we used to see earlier in case of Gen10 and Gen10 Plus.
![image](https://github.com/ansible-collections/community.general/assets/44438079/d46902d9-e67b-44b1-9015-b30ebef19614)

Reply:
any updates on this?

Reply:
Just because the HPE Gen 10 system doesn't produce a Task doesn't mean that no systems will do that; clients need to be prepared for either outcome. I know of systems that produce tasks and require system restarts for the volume to be created. Perhaps we can make this an additional change later when we have users performing this type of operation on those systems; I have some ideas on how to handle this based on what we did for firmware updates.

Reply:
Sure, we can make this change later then.

Reply:
Resolving this conversation. Thanks!